src/operator/expand.ts

```diff
@@ -2,7 +2,7 @@ import {Observable} from '../Observable';
 import {Scheduler} from '../Scheduler';
 import {ExpandOperator} from './expand-support';
 
-export function expand<T, R>(project: (value: T, index: number) => Observable<R>,
+export function expand<T, R>(project: (value: T, index: number) => Observable<R> | Promise<R>,
```

Comment:
do we accept projection returns `promise`? It seems implementation does not checks if returned is promise and does not trying to convert it internally but directly subscribe.